src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddressTest.java

```diff
@@ -0,0 +1,139 @@
+
+            () -> assertThat(performerNameAndAddress)
+                .usingRecursiveComparison()
+                .isEqualTo(PerformerNameAndAddress.fromString("NAD+SLA+A2442389:902++DR J SMITH"))
+        );
+    }
+
+    @Test
+    void when_mappingSegmentObjectToEdifactString_expect_returnCorrectEdifactString() {
+        String expectedEdifactString = "NAD+SLA+++LONDON CITY HOSPITAL'";
+        String expectedPerformerEdifactString = "NAD+SLA+A2442389:902++DR J SMITH'";
+
+        PerformerNameAndAddress performingOrganisation = PerformerNameAndAddress.builder()
+            .identifier("")
+            .code(null)
+            .performingOrganisationName("LONDON CITY HOSPITAL")
+            .partyName("")
+            .build();
+
+        PerformerNameAndAddress performer = PerformerNameAndAddress.builder()
+            .identifier("A2442389")
+            .code(HealthcareRegistrationIdentificationCode.CONSULTANT)
+            .performingOrganisationName("")
+            .partyName("DR J SMITH")
+            .build();
+
+        assertAll(
+            () -> assertEquals(expectedEdifactString, performingOrganisation.toEdifact()),
+            () -> assertEquals(expectedPerformerEdifactString, performer.toEdifact())
+        );
+    }
+
+    @Test
+    void when_mappingSegmentObjectToEdifactStringWithEmptyField_expect_edifactValidationExceptionIsThrown() {
+        PerformerNameAndAddress performingOrganisation = PerformerNameAndAddress.builder()
+            .identifier("")
+            .code(null)
+            .performingOrganisationName("")
+            .partyName("")
+            .build();
+
+        PerformerNameAndAddress performerWithoutName = PerformerNameAndAddress.builder()
+            .identifier("A2442389")
+            .code(HealthcareRegistrationIdentificationCode.CONSULTANT)
+            .performingOrganisationName("")
+            .partyName("")
+            .build();
+
+        assertAll(
+            () -> assertThrows(EdifactValidationException.class, performingOrganisation::toEdifact),
+            () -> assertThrows(EdifactValidationException.class, performerWithoutName::toEdifact)
+        );
+    }
+
+    @Test
+    void when_buildingSegmentObjectWithEmptyHealthcareCode_expect_noSuchElementExceptionIsThrown() {
+        assertThrows(
+            NoSuchElementException.class,
+            () -> PerformerNameAndAddress.builder()
+                .identifier("A2442389")
+                .code(HealthcareRegistrationIdentificationCode.fromCode(""))
+                .performingOrganisationName("")
+                .partyName("")
+                .build()
+        );
+    }
+
+    @Test
+    void when_buildingSegmentObjectWithoutMandatoryField_expect_nullPointerExceptionIsThrown() {
+        assertThrows(NullPointerException.class, () -> PerformerNameAndAddress.builder().build());
+    }
+
+    @Test
+    void testGetKey() {
+        assertEquals(performingOrganisationNameAndAddress.getKey(), "NAD");
```

Comment:
Expectation first for the basic `assertEquals`
```suggestion
        assertEquals("NAD", performingOrganisationNameAndAddress.getKey());
```

Reply:
Done in 3e4d259